stdlib.sh

```diff
@@ -1068,6 +1068,33 @@ layout_pyenv() {
   [[ -n "$PYENV_VERSION" ]] && export PYENV_VERSION
 }
 
+# Usage: layout uv
+#
+# Enables the uv project layout in the current directory, and syncs
+# the dependencies in the project.
+#
+# This relies on the `uv` command being available in the PATH, and performs a
+# sync on cd because uv is fast enough it's not impactful. It relies on uv's
+# configuration file and environment variables, rather than arguments.
+#
+layout_uv() {
+  # watch the uv configuration file for changes
+  watch_file .python-version
+  watch_file pyproject.toml
+  watch_file uv.lock
+
+  uv sync || return 1
+
+  # activate the virtualenv after syncing; this puts the newly-installed
+  # binaries on PATH.
+  venv_path=$(pwd)/.venv
+  if [[ -f $venv_path ]]; then
+    # shellcheck source=/dev/null
+    source "$venv_path/bin/activate"
```

Comment:
Does this work for all shells and platforms? (e.g. direnv supports Windows, albeit with issues)

Reply:
I think it should work on Linux, Mac and all Unix-like operating systems, which is what direnv supports: https://direnv.net/#prerequisites

Except for our comment previously about using `UV_PROJECT_ENVIRONMENT`.

Reply:
> I think it should work on Linux, Mac and all Unix-like operating systems, which is what direnv supports: [direnv.net#prerequisites](https://direnv.net/#prerequisites)

I've seen that line in the README, but it's 5+ years old and doesn't square with the fact that there have been releases for Windows in the meantime and that direnv is on winget. 🤷🏻‍♂️ 

That said, I didn't manage to get direnv to run on Windows out-of-the-box, so the pragmatic view is probably that it doesn't matter much?